src/gov/nist/javax/sip/stack/SIPTransactionStack.java

```diff
@@ -3292,6 +3292,22 @@ public boolean closeReliableConnection(String myAddress, int myPort, String tran
         return false;
     }
 
+	public Map<String, SIPDialog> getEarlyDialogTable() {
+		return earlyDialogTable;
+	}
+
+	public void setEarlyDialogTable(Map<String, SIPDialog> earlyDialogTable) {
```

Comment:
Totally agree with the concern and adding a comment, but again, once exposed the responsibility shifts away from us and the app developer is responsible for thread safety. Java doesn't offer much to enforce the rule so we don't have a good choice.

Reply:
Didn't knew that `ConcurrentMap` existed at first, know I know it !
I can make another PR to set `ConcurrentMap` instead `Map`.

Reply:
The objective here is to give the user of the code the most chances to avoid bugs while also allowing for a good amount of flexibility. Using ConcurrentMap give s a very good hint to the use what to provide but might take too much of the flexibility away (not sure how many of the popular map implementations extend ConcurrentMap). Anyways, setting a map with a clear comment is acceptable for me.

Reply:
@jagliot Totally agree that it is a perfect fit for `ConcurrentMap`, at first I didn't knew that it existed. For example MapDB provides a `ConcurrentMap` which is perfect. I will do a PR to set `ConcurrentMap` instead of `Map`.